src/main/java/spoon/support/compiler/ZipFolder.java

```diff
@@ -57,22 +65,47 @@ public List<SpoonFile> getFiles() {
 		// Indexing content
 		if (files == null) {
 			files = new ArrayList<>();
-			try (ZipInputStream zipInput = new ZipInputStream(new BufferedInputStream(new FileInputStream(file)));
-				ByteArrayOutputStream output = new ByteArrayOutputStream(2048)) {
-
-				ZipEntry entry;
-				while ((entry = zipInput.getNextEntry()) != null) {
-					zipInput.transferTo(output);
-					files.add(new ZipFile(this, entry.getName(), output.toByteArray()));
-					output.reset();
+			try (FileSystem zip = FileSystems.newFileSystem(URI.create("jar:" + file.toURI()), Map.of())) {
+				Path tempFolder = Files.createTempDirectory("spoon-zip-file-proxy");
+				// Try to clean up - not guaranteed to work!
```

Comment:
This wasn't guaranteed before as well and depends on the mood of the JVM. `File#deleteOnExit()` which was used before in Sources also registers a shutdown hook.